src/extensions/scratch3_ev3/index.js

```diff
@@ -0,0 +1,921 @@
+            this.speed * -1,
+            BTCommand.LONGRAMP
+        ));
+
+        // Send message
+        this._bt.sendMessage({
+            message: Base64Util.arrayBufferToBase64(cmd),
+            encoding: 'base64'
+        });
+
+        // Yield for time
+        return new Promise(resolve => {
+            setTimeout(() => {
+                resolve();
+            }, time);
+        });
+    }
+
+    motorRotate (port, degrees) {
+        if (!this.connected) return;
+
+        // TODO: Build up motor command
+        log.info(`motor rotate port: ${port} and degrees: ${degrees}`);
+    }
+
+    motorSetPosition (port, degrees) {
+        if (!this.connected) return;
+
+        // TODO: Build up motor command
+        log.info(`motor set position port: ${port} and degrees: ${degrees}`);
+    }
+
+    motorSetPower (port, power) {
+        if (!this.connected) return;
+
+        // TODO: Build up motor command
+        log.info(`motor set power port: ${port} and degrees: ${power}`);
+    }
+
+    _applyPrefix (n, cmd) {
+        const len = cmd.length + 5;
+        return [].concat(
+            len & 0xFF,
+            (len >> 8) & 0xFF,
+            0x1,
+            0x0,
+            0x0,
+            n,
+            0x0,
+            cmd
+        );
+    }
+
+    /**
+     * Generate a motor command in EV3 byte array format (CMD, LAYER, PORT,
+     * SPEED, RAMP UP, RUN, RAMP DOWN, BREAKING TYPE)
+     * @param  {string} command Motor command primitive (i.e. "prefix")
+     * @param  {string} port    Port to address
+     * @param  {number} n       Value to be passed to motor command
+     * @param  {number} speed   Speed value
+     * @param  {number} ramp    Ramp value
+     * @return {array}          Byte array
+     */
+    _motorCommand (command, port, n, speed, ramp) {
+        /**
+         * Generate run values for a given input.
+         * @param  {number} run Run input
+         * @return {array}      Run values (byte array)
+         */
+        const getRunValues = function (run) {
+            // If run duration is less than max 16-bit integer
+            if (run < 0x7fff) {
+                return [
+                    BTCommand.NUM16,
+                    run & 0xff,
+                    (run >> 8) & 0xff
+                ];
+            }
+
+            // Run forever
+            return [
+                BTCommand.NUM32,
+                run & 0xff,
+                (run >> 8) & 0xff,
+                (run >> 16) & 0xff,
+                (run >> 24) & 0xff
+            ];
+        };
+
+        // If speed is less than zero, make it positive and multiply the input
+        // value by -1
+        if (speed < 0) {
+            speed = -1 * speed;
+            n = -1 * n;
+        }
+
+        // If the input value is less than 0
+        const dir = (n < 0) ? 0x100 - speed : speed; // step negative or possitive
+        n = Math.abs(n);
+
+        // TODO ^^^^^ Is all of this shit really needed? ^^^^^ - ascii
```

Comment:
It looks like this comment started out in the prototype code, but I'd like to keep the language in our official repositories clean :)